app/utils/push_notifications.js

```diff
@@ -144,23 +144,21 @@ class PushNotificationUtils {
                         channel_id: data.channel_id,
                     },
                 });
-                completed();
+                completion();
                 return;
             }
 
-            if (badge >= 0) {
-                PushNotifications.setApplicationIconBadgeNumber(badge);
-            }
-
-            dispatch(markChannelViewedAndRead(data.channel_id));
             this.replyNotificationData = null;
-            completed();
+
+            PushNotifications.getDeliveredNotifications((notifications) => {
+                PushNotifications.setApplicationIconBadgeNumber(notifications.length);
```

Comment:
this may not be true as the notification center may not have notifications as there are two use cases:
1. manually clear the notification from the notification center.
2. in-app notifications do not add them to the notification center.
3. the reply fails to send a we add a new notification in the notification center, informing the user that the reply failed.

I'm thinking that here we may want to perform a request or maybe even use the selector to get the badges or something like that.

Problems with the above suggestion:
* Fetch from the server:
    - connection issues
    - server still includes the number of mentions of the current one being replied
* Use the info in the redux store
    - store still includes the number of mentions of the current one being replied

Reply:
We could try and solve it here, or if you think is out of scope then lets create another ticket for this. if the later please attach ticket.